inference-engine/tests/functional/plugin/cpu/shared_tests_instances/single_layer_tests/scatter_ND_update.cpp

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <vector>
+#include <ngraph/opsets/opset3.hpp>
+
+#include "single_layer_tests/scatter_ND_update.hpp"
+#include "common_test_utils/test_constants.hpp"
+
+using namespace LayerTestsDefinitions;
+using namespace ngraph::opset3;
+
+namespace {
+const std::vector<InferenceEngine::Precision> inputPrecisions = {
+        InferenceEngine::Precision::FP32,
+        InferenceEngine::Precision::BF16,
+        InferenceEngine::Precision::I32,
+        InferenceEngine::Precision::I16,
+        InferenceEngine::Precision::U8,
+};
+
+const std::vector<InferenceEngine::Precision> idxPrecisions = {
+        InferenceEngine::Precision::I32,
+        InferenceEngine::Precision::I64,
+};
+
+const std::vector<std::vector<size_t>> inputShapes = {
+        {10, 16, 28, 28, 22}
+};
+
+// slice replacement
+const std::vector<std::vector<size_t>> idxShapes_Depth1 = {
+        {4, 1}
+};
+const std::vector<std::vector<size_t>> idxValue_Depth1 = {
+        {0, 2, 4, 6}
+};
+const std::vector<std::vector<size_t>> updateShapes_Depth1 = {
+        {4, 16, 28, 28, 22}
+};
+
+const std::vector<std::vector<size_t>> idxShapes_Depth2 = {
+        {4, 2}
+};
+const std::vector<std::vector<size_t>> idxValue_Depth2 = {
+        {0, 2, 4, 6, 5, 6, 7, 9}
+};
+const std::vector<std::vector<size_t>> updateShapes_Depth2 = {
+        {4, 28, 28, 22}
+};
+
+const std::vector<std::vector<size_t>> idxShapes_Depth3 = {
+        {2, 3}
+};
+const std::vector<std::vector<size_t>> idxValue_Depth3 = {
+        {1, 5, 3, 6, 3, 9}
+};
+const std::vector<std::vector<size_t>> updateShapes_Depth3 = {
+        {2, 28, 22}
+};
+
+const std::vector<std::vector<size_t>> idxShapes_Depth4 = {
+        {2, 4}
+};
+const std::vector<std::vector<size_t>> idxValue_Depth4 = {
+        {0, 2, 4, 6, 5, 6, 7, 9}
+};
+const std::vector<std::vector<size_t>> updateShapes_Depth4 = {
+        {2, 22}
+};
+
+// elements replacement
+const std::vector<std::vector<size_t>> idxShapes_Depth5 = {
+        {2, 5}
+};
+const std::vector<std::vector<size_t>> idxValue_Depth5 = {
+        {2, 4, 5, 5, 1, 7, 9, 7, 8, 5}
+};
+const std::vector<std::vector<size_t>> updateShapes_Depth5 = {
+        {2}
+};
+
+
+const auto ScatterNDUpdate_SLICE1 = ::testing::Combine(
+        ::testing::ValuesIn(inputShapes),
```

Comment:
IMO, this is such a complex design. You can make sure, that the similar case is in `squeeze_unsqueeze` single layer tests. I think, we can do the same here.

Reply:
Thank you. I will look into.

Reply:
@iefode I have combined and simplified, thank you!